src/operator/zip-support.ts

```diff
@@ -27,7 +27,7 @@ export class ZipSubscriber<T, R> extends Subscriber<T> {
   private index = 0;
   private values: any;
   private project: (...values: Array<any>) => R;
-  private iterators = [];
+  private iterators: LookAheadIterator<any>[] = [];
```

Comment:
`LookAheadIterator` defines `hasValue` and `hasCompleted`, used elsewhere with iterator values.